### PR TITLE
Circuit interface to create Keccak gadget

### PR DESCRIPTION
**Pull request by querolita**
Closes https://github.com/o1-labs/proof-systems/issues/1219 https://github.com/o1-labs/proof-systems/issues/1220

### CODE REVIEW

File: kimchi/src/circuits/polynomials/keccak/mod.rs [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+    0x000000000000800a,
+    0x800000008000000a,
+    0x8000000080008081,
+    0x8000000000008080,
+    0x0000000080000001,
+    0x8000000080008008,
+];
+
+// Takes a dense u64 word and decomposes it into a vector of 4 dense quarters
+pub(crate) fn decompose(word: u64) -> Vec<u64> {
```
Comment (dannywillems) [inline kimchi/src/circuits/polynomials/keccak/mod.rs:78]: I will sound annoying :): but tests with corner cases would have been nice to have, just for the reviewers to be convinced it is fine at a first sight.

File: kimchi/src/circuits/polynomials/keccak/mod.rs [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+    0x8000000000000080,
+    0x000000000000800a,
+    0x800000008000000a,
+    0x8000000080008081,
+    0x8000000000008080,
+    0x0000000080000001,
+    0x8000000080008008,
+];
+
+// Takes a dense u64 word and decomposes it into a vector of 4 dense quarters
```
Comment (dannywillems) [inline kimchi/src/circuits/polynomials/keccak/mod.rs:77]: Maybe specify the order in the doc

File: kimchi/src/circuits/polynomials/keccak/mod.rs [hunk 3]
```diff
@@ -0,0 +1,103 @@
+//! Keccak hash module
+pub mod circuitgates;
+pub mod gadget;
+
+pub const DIM: usize = 5;
+pub const QUARTERS: usize = 4;
+pub const ROUNDS: usize = 24;
+pub const RATE: usize = 1088 / 8;
```
Comment (dannywillems) [inline kimchi/src/circuits/polynomials/keccak/mod.rs:8]: Nit: I like having the unit in the variable name. For instance, `RATE_IN_BYTES`. It helps when you use it somewhere else. When you expose it, it creates a better API.

**Review by dannywillems — APPROVED** [review]
LGTM, except having small unit tests and nit comments on documentation and variable naming.

4qjnyaegv7ly
